plugins/woocommerce-admin/client/homescreen/index.tsx

```diff
@@ -50,10 +49,8 @@ const withSelectHandler = ( select: WCDataSelector ) => {
 };
 
 export default compose(
-	onboardingData.profile
-		? withOnboardingHydration( {
-				profileItems: onboardingData.profile,
-		  } )
-		: identity,
```

Comment:
Conditional composition here is a bit confusing and potentially prone to bugs so I moved the `onboardingData.profile` check into the HOC.